### ISSUE TITLE
JsonConverterAttribute cannot be applied to an Enum

### ISSUE THREAD

**Issue by Joelius300** [dotnet/corefx#39741]
Since I'm trying to migrate from `Newtonsoft.Json` to the new `System.Text.Json` APIs, I have to be able to use a Enum-string converter.  

In json.net you can do this really easily with this attribute on the enum.  
```
[Newtonsoft.Json.JsonConverter(typeof(Newtonsoft.Json.Converters.StringEnumConverter))]
```

Since there is both a [JsonConverterAttribute](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonconverterattribute?view=netcore-3.0) and a [JsonStringEnumConverter](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonstringenumconverter?view=netcore-3.0) class, I thought you would do it like this:
```
[System.Text.Json.Serialization.JsonConverter(typeof(System.Text.Json.Serialization.JsonStringEnumConverter))]
```

However after getting a compiler error I quickly realized that the `JsonConverterAttribute` can't even be used on an Enum because it only has the following three `AttributeTargets`-flags.  
 - `AttributeTargets.Class`
 - `AttributeTargets.Struct`
 - `AttributeTargets.Property`

Are you not supposed to use this attribute and converter like this?  
Are there any samples because there I've not found any in [the docs](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization?view=netcore-3.0)?

**Comment by marcmarcet** [comment]
I was able to do configure this globally in asp.net core without using attributes like this:

```
services.AddControllers().AddJsonOptions(options => options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter()));
```

I supose you should be able to do the same passing the same options to your JsonSerializer

**Comment by Joelius300** [comment]
I tried that. These options won't get used globally when using `JsonSerializer` (which is very understandable) but also won't get used when using `IJSRuntime` (which is not at all understandable).  
I need a way of ignoring null values for `IJSRuntime` which uses `System.Text.Json` and does not expose any possibility to define options or customize the serializer. The first thought would be one of the `JsonAttribute`s since these will be respected no matter from where and with what options the serializer gets called.

**Comment by steveharter** [comment]
This does look to be an oversight on this scenario. I'll bring up a PR soon on this and review it internally for consideration for 3.0.

cc @JeremyKuhne

**Comment by Joelius300** [comment]
Thank you @steveharter. Will this fix be available in one of the next previews and if so in which one? Or do you also do fixes that will only be released on the 3.0 stable release?

**Comment by karelz** [comment]
Fixed in master (5.0) in PR #39790 and also in release/3.0 branch (3.0) in PR #39812.
